app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php

```diff
@@ -208,10 +208,10 @@ public function saveAction()
 
             //validate attribute_code
             if (isset($data['attribute_code'])) {
-                $validatorAttrCode = new Zend_Validate_Regex(array('pattern' => '/^[a-z][a-z_0-9]{1,254}$/'));
+                $validatorAttrCode = new Zend_Validate_Regex(array('pattern' => '/[a-z][a-z_0-9]{1,254}[^event]/'));
```

Comment:
This will match any string containing only characters in the set 'evnt' which may not spell "event", such as "ten", "vent", etc.

This does not need to be complicated, it should simply be a separate if clause with a string comparison like:

```php
if ('event' == $data['attribute_code']) {
    ...
}
```

That would have the added benefit of not breaking existing translations for the original message.

Reply:
Will not help much due it is magento core. If your PR mismatches from official magento PR it will be rejected, even if the fixes very good cause

Reply:
Yeah, just noting it here for people that use this repo. Also, the regex change broke the validation that the string must start with a character so now other characters can be injected into the attribute code like "foo.bar/abc" will now validate. Whoever wrote that regex obviously doesn't know anything about regex...

Reply:
it doesnt matter as the official magento core have same issue. Please create bug report for magento. This is not correct place.

Reply:
@colinmollenhour I promise it wasn't us that wrote that regex :)

Reply:
Magento 1 does not have a github project so as far as I'm concerned there is nowhere to report bugs to other than magento-lts.

Reply:
As I told you, please create bug report for magento. This is not correct place. Only exposing this problem with technical details publicly will lead into abuse of this problem much much faster.

Reply:
> it doesn't matter

It could matter since this could be a security issue and some users may wish to hold off on this upgrade.

Reply:
@treestonemedia Yes, I understand, this repo is just a mirror and doesn't accept bug reports or fixes and is only meant to be a perfect copy of the latest download. 

@Niko9911 Will not do.

Reply:
An if it is fixed in this version, it would not be 1.9.4.0 and this would not be mirror. If you want fix it, please create fork and use it.

If PR doesn't match the official Magento provided on their site it should be denied. If mirror starts making of changes into code, I'm afraid but I cannot trust the mirror anymore. That kinda behavior can easily lead into injection of malicious code.

If official patch comes out, it can be applied, but custom shit? Nope.

Reply:
@colinmollenhour It's nice to talk security issues about Magento while not reporting them.

Reply:
As I told you, there is life outside of GitHub. You do not need repository to report it to them. Be smart, not a dummy who needs hand-to-hand tutorial.

Well, that still passed their review. Whatever, I'm done with you if you cannot use your keyboard to report it. Instead of that you're just writing about it in GitHub repositories.

Reply:
Here, will make it easy for you to report: https://regex101.com/r/dRFXLm/2

Reply:
@Niko9911 can we keep things civil without personal insults please?

Reply:
@davidwindell I'm not insulting, but I'm highly questioning why he cannot create bug report as normal people would do instead spamming about it on GitHub.